Lib/defcon/objects/base.py

```diff
@@ -398,6 +543,25 @@ def update(self, other):
             self.postNotification(self.updateNotificationName)
         self.dirty = True
 
+    # -----------------------------
+    # Serialization/Deserialization
+    # -----------------------------
+
+    def getDataForSerialization(self, **kwargs):
+        from copy import deepcopy
+
+        deep_get = lambda k: deepcopy(self[k])
+
+        getters = []
+        for k in self.keys():
+            k = deepcopy(k) # needed?
+            getters.append((k, deep_get))
```

Comment:
why do we need to `deepcopy` when serializing? Isn't it safe to assume serialization is a read-only process?